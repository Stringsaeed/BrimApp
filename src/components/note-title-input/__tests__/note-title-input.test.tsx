import { fireEvent, screen } from "@testing-library/react-native";
import { useFormikContext } from "formik";
import React from "react";
import { TextInput } from "react-native";

import render from "utils/test";

import NoteTitleInput from "../";

// Mock the useFormikContext hook
jest.mock("formik", () => ({
  useFormikContext: jest.fn(),
}));

describe("NoteTitleInput", () => {
  const mockHandleChange = jest.fn();
  const mockHandleBlur = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    // Setup default mock implementation
    (useFormikContext as jest.Mock).mockReturnValue({
      values: { title: "Test Title" },
      handleChange: mockHandleChange,
      handleBlur: mockHandleBlur,
    });
  });

  it("renders correctly", () => {
    const { toJSON } = render(<NoteTitleInput />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("displays the title from form context", () => {
    render(<NoteTitleInput />);
    expect(screen.getByDisplayValue("Test Title")).toBeTruthy();
  });

  it("shows placeholder when title is empty", () => {
    (useFormikContext as jest.Mock).mockReturnValue({
      handleChange: mockHandleChange,
      handleBlur: mockHandleBlur,
      values: { title: "" },
    });

    render(<NoteTitleInput />);
    expect(
      screen.getByPlaceholderText("Wanna title your note? ðŸ˜’")
    ).toBeTruthy();
  });

  it("calls handleChange when text changes", () => {
    render(<NoteTitleInput />);
    fireEvent.changeText(
      screen.getByLabelText("Text input field"),
      "New Title"
    );
    expect(mockHandleChange).toHaveBeenCalledWith("title");
  });

  it("calls handleBlur when input loses focus", () => {
    render(<NoteTitleInput />);
    fireEvent(screen.getByLabelText("Text input field"), "blur");
    expect(mockHandleBlur).toHaveBeenCalledWith("title");
  });

  it("accepts and applies custom props", () => {
    render(<NoteTitleInput testID="custom-title-input" />);
    expect(screen.getByTestId("custom-title-input")).toBeTruthy();
  });

  it("forwards ref correctly", () => {
    const ref = React.createRef<TextInput>();
    render(<NoteTitleInput ref={ref} />);
    expect(ref.current).not.toBeNull();
  });
});
