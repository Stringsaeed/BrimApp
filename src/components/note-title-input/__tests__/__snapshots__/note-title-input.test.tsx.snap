// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteTitleInput renders correctly 1`] = `
<TextInput
  accessibilityLabel="Text input field"
  onChangeText={[Function]}
  placeholder="Wanna title your note? ðŸ˜’"
  placeholderTextColor="hsl(0, 0%, 56.1%)"
  style={
    {
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderStyle": "solid",
      "borderTopWidth": 0,
      "color": "hsl(0, 0%, 9.0%)",
      "fontSize": 20,
      "fontWeight": "bold",
      "paddingBottom": 10,
      "paddingLeft": 13,
      "paddingRight": 13,
      "paddingTop": 10,
    }
  }
  textAlignVertical="auto"
  value="Test Title"
/>
`;
